src/compiler/manipulate_wasm_binary.ts

```diff
@@ -0,0 +1,244 @@
+            addCanisterMethod(
+                module,
+                `${
+                    composite === true
+                        ? 'canister_composite_query'
+                        : 'canister_query'
+                } ${functionName}`,
+                'execute_js',
+                index,
+                true
+            );
+        }
+    );
+
+    compilerInfo.canister_methods.updates.forEach(
+        ({ name: functionName, index }) => {
+            addCanisterMethod(
+                module,
+                `canister_update ${functionName}`,
+                'execute_js',
+                index,
+                true
+            );
+        }
+    );
+
+    addCanisterMethod(
+        module,
+        'canister_init',
+        'init',
+        compilerInfo.canister_methods.init?.index ?? -1,
+        true
+    );
+
+    addCanisterMethod(
+        module,
+        'canister_post_upgrade',
+        'post_upgrade',
+        compilerInfo.canister_methods.post_upgrade?.index ?? -1,
+        true
+    );
+
+    if (compilerInfo.canister_methods.pre_upgrade !== undefined) {
+        addCanisterMethod(
+            module,
+            'canister_pre_upgrade',
+            'execute_js',
+            compilerInfo.canister_methods.pre_upgrade.index,
+            false
+        );
+    }
+
+    if (compilerInfo.canister_methods.inspect_message !== undefined) {
+        addCanisterMethod(
+            module,
+            'canister_inspect_message',
+            'execute_js',
+            compilerInfo.canister_methods.inspect_message.index,
+            true
+        );
+    }
+
+    if (compilerInfo.canister_methods.heartbeat !== undefined) {
+        addCanisterMethod(
+            module,
+            'canister_heartbeat',
+            'execute_js',
+            compilerInfo.canister_methods.heartbeat.index,
+            false
+        );
+    }
+
+    const memoryInfo = module.getMemoryInfo();
+
+    const numMemorySegments = module.getNumMemorySegments();
+
+    let segments = [];
+
+    for (let i = 0; i < numMemorySegments; i++) {
+        const segment = module.getMemorySegmentInfoByIndex(i);
+
+        segments.push(segment);
+    }
+
+    const normalizedSegments = segments.map((segment) => {
+        return {
+            offset: module.i32.const(segment.offset),
+            data: new Uint8Array(segment.data),
+            passive: segment.passive
+        };
+    });
+
+    const consumer = await getConsumer(canisterConfig);
+
+    const jsEncoded = new TextEncoder().encode(js);
+    const wasmDataEncoded = new TextEncoder().encode(
+        JSON.stringify({
+            env_vars: compilerInfo.env_vars,
+            consumer,
+            management_did: (
+                await readFile(
+                    join(AZLE_PACKAGE_PATH, 'canisters', 'management', 'ic.did')
+                )
+            ).toString()
+        })
+    );
+
+    module.setMemory(memoryInfo.initial, memoryInfo.max ?? -1, null, [
```

Comment:
I'm assuming that -1 is how you signify there is no max. I am confused at how the max for the memory before we manipulate it would be the same for the memory after we manipulate it, and what would happen if we go beyond that max